plugins/inventory/libvirt.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+name: libvirt
+plugin_type: inventory
+extends_documentation_fragment:
+    - constructed
+short_description: Libvirt inventory source
+description:
+    - Get libvirt guests in an inventory source
+author:
+    - Dave Olsthoorn <dave@bewaar.me>
+version_added: "2.10"
+options:
+    plugin:
+        description: token that ensures this is a source file for the 'libvirt' plugin.
+        required: True
+        choices: ['libvirt']
+    connection:
+        description: Libvirt Connection URI
+        required: True
+        type: str
+requirements:
+    - "libvirt"
+'''
+
+EXAMPLES = r'''
+# Connect to lxc host
+plugin: libvirt
+connection: 'lxc:///'
+
+# Connect to qemu host
+plugin: libvirt
+connection: 'qemu:///system'
+'''
+
+import libvirt
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+    NAME = 'libvirt'
+
+    def parse(self, inventory, loader, path, cache=True):
+        super(InventoryModule, self).parse(
+            inventory,
+            loader,
+            path,
+            cache=cache
+        )
+
+        conn = libvirt.open(self.get_option('connection'))
+        if not conn:
+            raise Exception("hypervisor connection failure")
```

Comment:
Exceptions should be raised via AnsibleError, so with

from ansible.errors import AnsibleError

you can raise the error like so:

raise AnsibleError("hypervisor connection failure")